controller/auth-controller.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const debug = require('debug')('ramble:auth-controller');
+const User = require('../model/user');
+const httpErrors = require('http-errors');
+
+exports.signup = function(reqBody){
+  return new Promise((resolve, reject) => {
+    debug('signupPromise');
+    var password = reqBody.password;
+    delete reqBody.password;
+    var user = new User(reqBody);
+    user.generateHash(password)
+    .then(user => user.save())
+    .then(user => user.generateToken())
+    .then(token => resolve(token))
+    .catch(reject);
+  });
+};
+
+exports.signin = function(auth){
+  debug('signinPromise');
+  return new Promise((resolve, reject) => {
+    User.findOne({username:auth.username})
+      .then((user) => {
+        if(user === null){
+          return reject(httpErrors(401, 'wrong username or password'));
+        }
+        return user.compareHash(auth.password);
+      })
+      .then(user =>  user.generateToken())
+      .then(token => resolve(token))
+      .catch(reject);
+  });
+};
+
+
+// post with a token
+// and body should contain
+//   oldPassword
+//   newPassword
+exports.updatePassword = function(userId, reqBody){
```

Comment:
Great work creating this update password, logicly i dont see any flaws with it This was a great streach goal!!!